js/app.js

```diff
@@ -206,29 +207,26 @@ var TwineApp = Backbone.Marionette.Application.extend(
       }, { wait: true });
 
       // and child passages
-      
-      $story.find(selectors.passageData).each(function()
-      {
+
+      var createPassagesFromData = function() {
         var $passage = $(this);
         var id = $passage.attr('pid');
         var pos = $passage.attr('position');
         var posBits = pos.split(',');
-        var tags = $passage.attr('tags').trim();
+        var tags = $passage.attr('tags') ? $passage.attr('tags').trim() : "";
         tags = tags === "" ? [] : tags.split(/\s+/);
 
-        var passage = allPassages.create(
-        {
-          name: $passage.attr('name'),
-          tags: tags,
-          text: $passage.text(),
-          story: story.id,
-          left: parseInt(posBits[0]),
-          top: parseInt(posBits[1])
-        }, { wait: true }); 
-
-        if (id == startPassageId)
-          story.save({ startPassage: passage.id });
-      });
+        var passageObj = createPassage($passage, story.id);
+
+        var passageModel = allPassages.create(passageObj, { wait: true });

```

Comment:
Still confused what's going on this line. So this essentially is a backbone call for: 

`Passage.collection.all().create(passageObj, {wait: true})`

My questions: 

1. Why is the `.all()` call necessary? Can't find an example of that online. 
2. How is Passage smart enough to know which type of passage to create?